src/main/java/mate/academy/springboot/datajpa/service/CategoryServiceImpl.java

```diff
@@ -0,0 +1,39 @@
+package mate.academy.springboot.datajpa.service;
+
+import mate.academy.springboot.datajpa.model.Category;
+import mate.academy.springboot.datajpa.repository.CategoryRepository;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository repository;
+
+    public CategoryServiceImpl(CategoryRepository repository) {
+        this.repository = repository;
+    }
+
+    @Override
+    public Category save(Category category) {
+        return repository.save(category);
+    }
+
+    @Override
+    public Category get(Long id) {
+        return repository.getById(id);
+    }
+
+    @Override
+    public void delete(Long id) {
+        repository.deleteById(id);
+    }
+
+    @Override
+    public void update(Category category, Long id) {
+        if (!repository.getById(id).equals(null)) {
+            category.setId(id);
+            repository.save(category);
+            return;
+        }
+        throw new RuntimeException("Category by id " + id + " doesn't exist");
```

Comment:
```suggestion
            repository.save(category);
```

I think we could just use save method here